prettyprint_test.go

```diff
@@ -0,0 +1,77 @@
+package sequitur
+
+import (
+	"bytes"
+	"fmt"
+)
+
+const testString = `
+pease porridge hot,
+pease porridge cold,
+pease porridge in the pot,
+nine days old.
+
+some like it hot,
+some like it cold,
+some like it in the pot,
+nine days old.
+
+《施氏食狮史》
+石室诗士施氏，嗜狮，誓食十狮。
+氏时时适市视狮。
+十时，适十狮适市。 
+是时，适施氏适市。氏视是十狮，恃矢势，使是十狮逝世。
+氏拾是十狮尸，适石室。
+石室湿，氏使侍拭石室。
+石室拭，氏始试食是十狮尸。
+食时，始识是十狮，实十石狮尸。
+试释是事。
+` // https://www.duolingo.com/comment/25323797/Fun-Chinese-poem-The-Lion-Eating-Poet-in-the-Stone-Den
+
+func ExamplePrettyPrint() {
+
+	g := new(Grammar)
+
+	err := g.Parse([]byte(testString))
+	if err != nil {
+		panic(err)
+	}
+
+	var output bytes.Buffer
+	err = g.PrettyPrint(&output)
+	if err != nil {
+		panic(err)
+	}
+
+	// this next line just required to make the output correct for the test to pass
+	fmt.Println(string(bytes.Replace(output.Bytes(), []byte(" \n"), []byte("\n"), -1)))
```

Comment:
or fix the code so that it doesn't produce a space at the end of each line by printing the `->` without a trailing space and changing prettyPrinter.print to print the space before, not after, each token?

Reply:
Will do.